src/primitives/__tests__/UnionBuilder.spec.js

```diff
@@ -77,4 +77,32 @@ describe('UnionBuilder', () => {
       expect(() => UnionBuilder.setField(TextBuilder)).to.throw();
     });
   });
+
+  describe('isEqual', () => {
+    it('can compare inequivalent builders', () => {
+      const international = StructBuilder
+        .setField('country', TextBuilder);
+      const usa = StructBuilder
+        .setField('country', TextBuilder)
+        .setField('state', TextBuilder);
+      const countriesBuilder = UnionBuilder
+        .setUnion([international, usa])
+        .setDispatch(value => (
+          value.country === 'usa' ? usa : international
+        ));
+
+      const otherInternational = StructBuilder
+        .setField('country', TextBuilder);
+      const otherUsa = StructBuilder
+        .setField('country', TextBuilder)
+        .setField('state', TextBuilder);
+      const otherCountriesBuilder = UnionBuilder
+        .setUnion([otherInternational, otherUsa])
+        .setDispatch(value => (
+          value.country === 'usa' ? otherUsa : otherInternational
+        ));
+
+      expect(countriesBuilder.isEqual(otherCountriesBuilder)).to.be.true;
```

Comment:
Can you add a test case where two unionbuilders are not equal? Perhaps where their dispatch is not the same.